### PR TITLE
Make install script(s) fail on unsupported systems

### PR DESCRIPTION
**Pull request by joakimr-axis**
Add check not only if to use Darwin or Linux, but also make sure not to
install the Linux amd64 binaries on e.g. a Linux armv7l system. Then a
proper (?) error message will be displayed and the script aborts.
When/if more systems are supported, they are easily added with this
structure.

Change-Id: Ia0b23785cb89a6620fcbba27aff8270bcf4008ff
Signed-off-by: Joakim Roubert <joakimr@axis.com>

### CODE REVIEW

**Comment by joakimr-axis** [comment]
With today's HEAD install script(s), this is what happens if run on a RPI/Raspbian:

```
$ uname -s
Linux
$ uname -m
armv7l
wget -qO- https://github.com/linkerd/linkerd2/releases/download/stable-2.5.0/linkerd2-cli-stable-2.5.0-linux |file -
/dev/stdin: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=z7NS7Xt8pSzY1sjvB4bN/u6428gc5CK2Z7PwCNqVq/aH1RWi-rz1U4vjP_H4ft/2yfIyQIif--v5lYoeL2b, stripped

```

File: run.linkerd.io/public/install [hunk 1]
```diff
@@ -4,11 +4,27 @@ set -eu
...
+      x86_64)
+        ;;
+      *)
+        echo "There is no linkerd $OS support for $arch" >&2
+        exit 1
+        ;;
+    esac
+    ;;
+  *)
+    echo "There is no linkerd support for $OS/$arch (yet...)" >&2
```
Comment (grampelberg) [inline run.linkerd.io/public/install]: How about `Please see https://github.com/linkerd/linkerd2/releases to download the correct binary for your platform.`? There's always the windows folks =)

Comment (joakimr-axis) [reply run.linkerd.io/public/install]: Will the Windows folks run this script?

Comment (grampelberg) [reply run.linkerd.io/public/install]: The docs don't differentiate, so I assume so? I've not found one of these mysterious windows folks to chat with =)

Comment (joakimr-axis) [reply run.linkerd.io/public/install]: With the current state of HEAD, the script would then download the linux amd64 version which is probably not what the Windows users would like.

But I will do some experiments on my Windows machine. Do we think the Windows people (yet to be found) are running it in Cygwin, PowerShell or native (?) bash?

Comment (joakimr-axis) [reply run.linkerd.io/public/install]: Ok, now I have made some Windows experiments too. The Windows people using the Linux Subsystem will have a system that presents itself (uname) as Linux/x86_64 just like a native Linux system. Using linkerd's linux binary then works.

I also tried it out with Cygwin, which presents (uname) itself as namely CYGWIN, and using the windows.exe there works well, so I will add that to my patch.

**Comment by grampelberg** [comment]
@joakimr-axis check out [contributing](https://github.com/linkerd/website/blob/master/CONTRIBUTING.md#certificate-of-origin) to get the checks passing. I can merge the PR after that =)

**Comment by joakimr-axis** [comment]
Ok, DCO fixed but there seems to be some internal error in the ci/circleci:test:

```
========================================================================
2017/11/20/bringing-service-communication-shadows-part-2/index.html
  Non-OK status: 404 --- 2017/11/20/bringing-service-communication-shadows-part-2/index.html --> https://istio.io/docs/tasks/telemetry/using-istio-dashboard/
2018/10/23/a-linkerd-2-0-deep-dive-unboxing-debugging-and-monitoring-kubernetes-services-on-azure/index.html
  Get https://github.com/BuoyantIO/emojivoto: dial tcp 140.82.114.4:443: connect: connection refused --- 2018/10/23/a-linkerd-2-0-deep-dive-unboxing-debugging-and-monitoring-kubernetes-services-on-azure/index.html --> https://github.com/BuoyantIO/emojivoto
========================================================================
```

**Pull request merged**